instrumented-tests/http-server-mock/Sources/HTTPServerMock/ServerSession.swift

```diff
@@ -44,4 +44,32 @@ public class ServerSession {
                 )
             }
     }
+
+    /// Actively fetches 'POST` requests recorded by the server until a desired count is found, or timeouts returning current recorded requests
+    public func pullRecordedPOSTRequests(count: Int, timeout: TimeInterval) throws -> [POSTRequestDetails] {
+        var currentRequests = [ServerMock.RequestInfo]()
+
+        let timeoutTimer = DispatchSource.makeTimerSource(queue: DispatchQueue.global())
+        timeoutTimer.setEventHandler { timeoutTimer.cancel() }
+        timeoutTimer.schedule(deadline: .now() + timeout, leeway: .nanoseconds(0))
+        if #available(iOS 10.0, *) {
+            timeoutTimer.activate()
+        }
```

Comment:
is timer automatically activated in macOS? why do we need `if #available`?

Reply:
No, the HTTP mock server should only run on macOS not on iOS, I changed the package for that and also for a minimum macOS version. But Xcode still shows iOS as a destination for the Scheme, so I added that line to avoid the Xcode error if you just try to build the library and you have an iPhone selected in Xcode.